wagtail/images/tests/tests.py

```diff
@@ -412,6 +437,12 @@ def test_get_with_extra_component(self):
         self.assertEqual(response.status_code, 200)
         self.assertTrue(response.streaming)
         self.assertEqual(response["Content-Type"], "image/png")
+        # Ensure the file can actually be read
+        with BytesIO() as buffer:
+            for _bytes in response:
+                buffer.write(_bytes)
+            image = willow.Image.open(buffer)
+        self.assertIsInstance(image, PNGImageFile)
```

Comment:
Nit: I don't think you need the loop here, although I may be wrong.
```suggestion
        # Ensure the file can actually be read
        image = willow.Image.open(response.content)
        self.assertIsInstance(image, PNGImageFile)
```

Reply:
Tried with `willow.Image.open(response.streaming_content)` but it's not quite a file, it's just a generator that can be consumed, so I got this error
```
Traceback (most recent call last):
  File "/home/sebcorbin/projects/wagtail/wagtail/images/tests/tests.py", line 478, in test_get_with_serve_action
    image = willow.Image.open(response.streaming_content)
  File "/home/sebcorbin/.virtualenvs/wagtail/lib/python3.10/site-packages/willow/image.py", line 84, in open
    image_format = filetype.guess_extension(f)
  File "/home/sebcorbin/.virtualenvs/wagtail/lib/python3.10/site-packages/filetype/filetype.py", line 63, in guess_extension
    kind = guess(obj)
  File "/home/sebcorbin/.virtualenvs/wagtail/lib/python3.10/site-packages/filetype/filetype.py", line 28, in guess
    return match(obj) if obj else None
  File "/home/sebcorbin/.virtualenvs/wagtail/lib/python3.10/site-packages/filetype/match.py", line 30, in match
    buf = get_bytes(obj)
  File "/home/sebcorbin/.virtualenvs/wagtail/lib/python3.10/site-packages/filetype/utils.py", line 84, in get_bytes
    raise TypeError('Unsupported type as file input: %s' % type(obj))
TypeError: Unsupported type as file input: <class 'map'>
```